pilot/pkg/keycertbundle/keycertbundle.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keycertbundle
+
+import (
+	"io/ioutil"
+	"sync"
+
+	"go.uber.org/atomic"
+)
+
+// KeyCertBundle stores the cert, private key and root cert for istiod.
+type KeyCertBundle struct {
+	CertPem  []byte
+	KeyPem   []byte
+	CABundle []byte
+}
+
+type Watcher struct {
+	// Indicated whether bundle has been set, it is used to invoke watcher for the first time.
+	initDone atomic.Bool
+	mutex    sync.Mutex
+	bundle   KeyCertBundle
+	watchers []chan KeyCertBundle
+}
+
+func NewWatcher() *Watcher {
+	return &Watcher{}
+}
+
+// AddWatcher returns channel to receive the updated items.
+func (w *Watcher) AddWatcher() chan KeyCertBundle {
+	ch := make(chan KeyCertBundle, 1)
+	w.mutex.Lock()
```

Comment:
Why does this function use w.mutex.Unlock() instead of "defer w.mutex.Unlock()" as other functions of Watcher?